common/build.gradle.kts

```diff
@@ -37,6 +41,31 @@ kotlin {
                 implementation(projects.kspAnnotations)
             }
         }
+
+        nativeMain {
+            dependencies {
+                // Native does not have compileOnly
+                implementation(projects.kspAnnotations)
```

Comment:
could you check again if this is needed? i.e. create another project depending on native kord and see if it'll compile (it doesn't for js if the implementation dependency is missing)